extension/agenthealth/handler/stats/provider/statuscode.go

```diff
@@ -0,0 +1,216 @@
+}
+
+// processStatusCode updates the stats map for the given status code entry.
+func (sp *StatsProvider) processStatusCode(entry statusCodeEntry) {
+	sp.mu.Lock()
+	defer sp.mu.Unlock()
+
+	stats, exists := sp.statsByOperation[entry.operation]
+	if !exists {
+		stats = &[5]int{}
+		sp.statsByOperation[entry.operation] = stats
+	}
+	sp.updateStatusCodeCount(stats, entry.statusCode)
+}
+
+// updateStatusCodeCount updates the count for the specific status code.
+func (sp *StatsProvider) updateStatusCodeCount(stats *[5]int, statusCode int) {
+	log.Printf("Updating stats for status code: %d\n", statusCode)
+	switch statusCode {
+	case 200:
+		stats[0]++
+	case 400:
+		stats[1]++
+	case 408:
+		stats[2]++
+	case 413:
+		stats[3]++
+	case 429:
+		stats[4]++
+	default:
+		log.Printf("Ignored status code: %d\n", statusCode)
+	}
+}
+
+// startResetTimer initializes a reset timer to clear stats periodically.
+func (sp *StatsProvider) startResetTimer() {
+	log.Println("Starting stats reset timer...")
+	sp.resetTimer = time.AfterFunc(statusResetInterval, func() {
+		sp.mu.Lock()
+		defer sp.mu.Unlock()
+		log.Println("Resetting stats...")
+		for key := range sp.statsByOperation {
+			delete(sp.statsByOperation, key)
+		}
+		sp.startResetTimer()
+	})
+}
+
+// StatusCodeHandler is the handler that uses the StatsProvider for processing.
+type StatusCodeHandler struct {
+	statsProvider *StatsProvider
+}
+
+func (h *StatusCodeHandler) ID() string {
+	return statusHandlerID
+}
+
+func (h *StatusCodeHandler) Position() awsmiddleware.HandlerPosition {
+	return awsmiddleware.After
+}
+
+// NewStatusCodeHandler creates a new handler with the given StatsProvider.
+func NewStatusCodeHandler(provider *StatsProvider) *StatusCodeHandler {
+	log.Println("Creating new StatusCodeHandler...")
+	return &StatusCodeHandler{statsProvider: provider}
+}
+
+// HandleResponse enqueues the status code into the StatsProvider's channel.
+func (h *StatusCodeHandler) HandleResponse(ctx context.Context, r *http.Response) {
+	operation := awsmiddleware.GetOperationName(ctx)
+	if !h.statsProvider.filter.IsAllowed(operation) {
+		return
+	}
+
+	operation = GetShortOperationName(operation)
+	if operation == "" {
+		return
+	}
+
+	h.statsProvider.EnqueueStatusCode(operation, r.StatusCode)
+}
+
+// GetShortOperationName maps long operation names to short ones.
+func GetShortOperationName(operation string) string {
+	switch operation {
+	case "PutRetentionPolicy":
+		return "pmd"
+	case "DescribeInstances":
+		return "di"
+	case "DescribeTags":
+		return "dt"
+	case "DescribeTasks":
+		return "dts"
+	case "DescribeVolumes":
+		return "dv"
+	case "DescribeContainerInstances":
+		return "dci"
+	case "DescribeServices":
+		return "ds"
+	case "DescribeTaskDefinition":
+		return "dtd"
+	case "ListServices":
+		return "ls"
+	case "ListTasks":
+		return "lt"
+	case "CreateLogGroup":
+		return "clg"
+	case "CreateLogStream":
+		return "cls"
+	default:
+		return ""
+	}
+}
```

Comment:
This function is essentially a duplicate of the string array `StatusCodeOperations`, so the two constructs need to be kept in-sync. I'd suggest either moving the function closer to the string array definition, or change the string array to a map which will do this mapping for you. Then, if you do need a list of valid operations, you can get it from the map.

Reply:
Also, did we decide to remove AssumeRole? It missing from this list and the StatusCodeOperations

Reply:
We need to get Assume role from the contrib changes so I was thinking of adding it then, but I can add it here too.